flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/MiniDispatcherTest.java

```diff
@@ -177,6 +177,34 @@ public void testTerminationAfterJobCompletion() throws Exception {
         }
     }
 
+    /**
+     * Tests that in detached mode, the {@link MiniDispatcher} will not complete the future that
+     * signals job termination if the JobStatus is not globally terminal state.
+     */
+    @Test
+    public void testNotTerminationWithoutGloballyTerminalState() throws Exception {
```

Comment:
the test passes without the fix as well

Reply:
I have modified the code to test the shutDownFuture not finished.